identity-service/src/app.js

```diff
@@ -121,9 +121,48 @@ class App {
     return whitelistRegex && !!ip.match(whitelistRegex)
   }
 
+  _getIP (req) {
+    // Gets the IP for ratelistening based on X-Forwarded-For headers
+    // Algorithm:
+    // If > 1 headers:
+    //    If rightmost is CN, then use rightmost - 1 (known to be safe)
+    //    If rightmost is not CN, then use rightmost
+    // If 1 header or no headers:
+    //    use req.ip (leftmost)
+
+    let ip = req.ip
+    const forwardedFor = req.get('X-Forwarded-For')
+
+    // This shouldn't ever happen since Identity will always be behind CF
+    if (!forwardedFor) {
+      console.debug('_getIP: no forwarded-for')
```

Comment:
should use req.logger instead of console so it's tracked along w/ the request

Reply:
Ah good catch